### PR TITLE
Add PQL support

### LINKED ISSUE #16: support for PQL in PuppetDB::Query
[PQL](https://docs.puppet.com/puppetdb/4.1/api/query/v4/pql.html) is a slightly more "natural" query language for most people

### PR DESCRIPTION
**Pull request by raphink**
Fixes #16

### CODE REVIEW

**Comment by raphink** [comment]
Rubocop fails on files I haven't touched.

File: spec/unit/client_spec.rb [hunk 1]
```diff
@@ -175,6 +175,25 @@ def expect_include_total(mock_query)
...
+
+    mock_response = mock
+    mock_response.expects(:code).returns(200)
+    mock_response.expects(:headers).returns('X-Records' => 0)
+    mock_response.expects(:parsed_response).returns([])
+
+    PuppetDB::Client.expects(:get).returns(mock_response).at_least_once.with do |_path, opts|
+      opts == {
+        body: {
+          'query'         => 'resources{}',
```
Comment (dhollinger) [inline spec/unit/client_spec.rb]: This is where one if the rubocop tests fail. There shouldn't be a comma after the last element of a hash.

Comment (raphink) [inline spec/unit/client_spec.rb]: Ack, fixed.

**Comment by dhollinger** [comment]
@raphink could you also add `puppetdb-ruby.gemspec` to the `.rubocop.yml` under the `Excludes` section?

That should fix the last rubocop failure (I thought I had fixed that a while ago).

**Comment by raphink** [comment]
@dhollinger don't and rebased to make both commits pass

**Comment by dhollinger** [comment]
One more request - We also need the `README` updated with documentation on how to use the new PQL feature.

**Comment by raphink** [comment]
Granted @dhollinger

**Comment by dhollinger** [comment]
@raphink thanks for the contribution!

**Pull request merged**